bindings/wasm/examples/editor.js

```diff
@@ -299,7 +302,25 @@ require(['vs/editor/editor.main'], async function() {
       addEdit(button);
     }
   }
-  switchTo(currentName);
+  const params = new URLSearchParams(window.location.search);
+  if (window.location.hash.length > 0) {
+    const name = unescape(window.location.hash.substring(1));
+    switchTo(name);
+  } else if (params.get('url') != null) {
+    console.log(`Fetching ${params.get('url')}`);
+    autoExecute = false;
+    const response = await fetch(params.get('url'));
+    const code = await response.text();
+    const name = uniqueName(params.get('name') ?? 'New Script');
+    setScript(name, code);
+    const nextButton = createDropdownItem(name);
+    newButton.insertAdjacentElement('afterend', nextButton.parentElement);
+    addEdit(nextButton);
+    switchTo(name);
```

Comment:
Is some of this duplicated code that could be a function?

Reply:
yes